src/scratchtocatrobat/converter/mediaconverter.py

```diff
@@ -81,8 +81,7 @@ def __init__(self, scratch_project, catrobat_program, images_path, sounds_path):
         self.sounds_path = sounds_path
         self.file_rename_map = {}
 
-
-    def convert(self, progress_bar = None):
+    def convert(self, progress_bar=None):
         all_used_resources = []
```

Comment:
By introducing a separate class that holds resource collections like all_used_resources, unconverted_media_resources or converted_media_resources_paths the list of function parameters could be reduced for most functions. Initializing it in the __init__ function of the MediaConverter class would make it accessible to underlying functions without having a parameter at all.